app/lib/analyzer/task_sources.dart

```diff
@@ -4,24 +4,79 @@
 
 import 'dart:async';
 
+import 'package:gcloud/db.dart';
+
+import '../frontend/models.dart';
 import '../shared/task_scheduler.dart';
 
+import 'models.dart';
+
+/// Creates a task when a package has no analysis for its latest stable version.
 class DatastoreHeadTaskSource extends PollingTaskSource {
-  DatastoreHeadTaskSource() : super(new Duration(minutes: 10));
+  final DatastoreDB _db;
+  DatastoreHeadTaskSource(this._db) : super(new Duration(minutes: 10));
 
   @override
   Stream<Task> pollTasks() {
-    // TODO: implement pollTasks
-    return new Stream.empty();
+    return _db
+        .query(Package)
```

Comment:
A few comments about `DatastoreHeadTaskSource` and `DatastoreHistoryTaskSource`

  * it would be fine if we ensured to analyze all package versions, not just the latest package.
  * it would be nice to not do repeated queries for things we know we have already processed.

So the idea with the split between `DatastoreHeadTaskSource` and `DatastoreHistoryTaskSource` was:

  * the head poller is just querying the latest `PackageVersion` uploads since the last poll (not going back in  history infinitely)
  * the history poller is just going back in history *once* and then it's done (i.e. can return an empty stream)

This should be easily doable via a query on `PackageVersion` with `-date` sorting and a field remembering, which was the last item we processed already, thereby making both `getTasks()` calls a NOP (after initial history traversal).